src/lib/datatypes/include/sol-memdesc.h

```diff
@@ -1085,6 +1086,525 @@ sol_memdesc_get_structure_member_memory(const struct sol_memdesc *structure_desc
 }
 
 /**
+ * @brief Helper to fetch the memory as the largest supported unsigned integer.
+ *
+ * @param desc the memory description.
+ * @param memory the memory to get content.
+ *
+ * @return the number as uint64_t. On errors, errno is set to non-zero.
+ *
+ * @see sol_memdesc_is_unsigned().
+ */
+static inline uint64_t
+sol_memdesc_get_as_uint64(const struct sol_memdesc *desc, const void *memory)
+{
+    int64_t i64;
+
+    errno = EINVAL;
+    if (!desc || !memory)
+        return 0;
+
+#ifndef SOL_NO_API_VERSION
+    if (desc->api_version != SOL_MEMDESC_API_VERSION_COMPILED)
+        return 0;
+#endif
+
+    errno = 0;
+    switch (desc->type) {
+    case SOL_MEMDESC_TYPE_UINT8:
+        return *(const uint8_t *)memory;
+    case SOL_MEMDESC_TYPE_UINT16:
+        return *(const uint16_t *)memory;
+    case SOL_MEMDESC_TYPE_UINT32:
+        return *(const uint32_t *)memory;
+    case SOL_MEMDESC_TYPE_UINT64:
+        return *(const uint64_t *)memory;
+    case SOL_MEMDESC_TYPE_ULONG:
+        return *(const unsigned long *)memory;
+    case SOL_MEMDESC_TYPE_SIZE:
+        return *(const size_t *)memory;
+    case SOL_MEMDESC_TYPE_INT8:
+        i64 = *(const int8_t *)memory;
+        goto check_signed;
+    case SOL_MEMDESC_TYPE_INT16:
+        i64 = *(const int16_t *)memory;
+        goto check_signed;
+    case SOL_MEMDESC_TYPE_INT32:
+        i64 = *(const int32_t *)memory;
+        goto check_signed;
+    case SOL_MEMDESC_TYPE_INT64:
+        i64 = *(const int64_t *)memory;
+        goto check_signed;
+    case SOL_MEMDESC_TYPE_LONG:
+        i64 = *(const long *)memory;
+        goto check_signed;
+    case SOL_MEMDESC_TYPE_SSIZE:
+        i64 = *(const ssize_t *)memory;
+        goto check_signed;
+    case SOL_MEMDESC_TYPE_BOOLEAN:
+        return *(const bool *)memory;
+    case SOL_MEMDESC_TYPE_DOUBLE:
+        i64 = *(const double *)memory;
+        goto check_signed;
```

Comment:
Do you care about `-0.0`? It will fail range checking below, but maybe it should pass.